test/unit/os/env_spec.lua

```diff
@@ -127,4 +129,66 @@ describe('env function', function()
       eq(hostname, (ffi.string(hostname_buf)))
     end)
   end)
+
+  function vim_setenv(name, value)
+    return env.vim_setenv((to_cstr(name)), (to_cstr(value)))
+  end
+
+  describe('vim_setenv', function()
+    local env_nvimruntime
+    local env_nvim
+
+    setup(function()
+      -- Save vars or '', nil is not acceptable and there is no unsetenv
```

Comment:
I don't think it's supported on Windows, but we could wrap the workaround (set variable to "")
https://stackoverflow.com/questions/3205197/remove-environmental-variable-programmatically